pkg/core/interop/context.go

```diff
@@ -71,6 +71,7 @@ func NewContext(trigger trigger.Type, bc Ledger, d *dao.Simple,
 	getContract func(*dao.Simple, util.Uint160) (*state.Contract, error), natives []Contract,
 	block *block.Block, tx *transaction.Transaction, log *zap.Logger) *Context {
 	baseExecFee := int64(DefaultBaseExecFee)
+	// TODO: we probably don't need to create one more wrapper if it's a TrieStore.
```

Comment:
We need it to handle writes if any,

Reply:
Sure, but the case is that TrieStore is already wrapped into MemCachedStore and from this state the new trie DAO is created. So technically, by this moment we already have pretty private and writable Store here.